File: SecondYearCS/OOP/Civitas/src/civitas/SorpresaIrCarcel.java
package civitas;

import java.util.ArrayList;

public class SorpresaIrCarcel extends Sorpresa{
    
    private Tablero tablero;     
    private int valor; 
    

    //se utiliza para mover a una casilla
    SorpresaIrCarcel(Tablero tablero, String texto){
        super(texto);
        this.tablero = tablero;
        this.valor = tablero.getCarcel();
    }
    
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if( this.jugadorCorrecto(actual, todos) ){
            if(todos.get(actual).encarcelar(this.valor)) //O tablero.getCarcel()
            this.informe(actual,todos);
        }
    }

    @Override
    public String toString(){
       return super.toString() + " casilla: " + valor;
    }

}

File: SecondYearCS/OOP/CivitasRB/lib/SorpresaIrCarcel.rb
require_relative 'sorpresa.rb'
module CivitasRB
  class SorpresaIrCarcel < Sorpresa
    attr_accessor :valor
    #Constructores por defecto y con parametro asdfas
    def initialize(tablero, texto)
      super(texto)
      @tablero = tablero
      @valor = tablero.numCasillaCarcel
    end

    #Informamos al diario y le enviamos a la casilla adecuada.
    def aplicarAJugador(act,todos)
      if self.jugadorCorrecto(act, todos)
        self.informe(act,todos)
        todos.at(act).encarcelar(@valor)
      end
    end

    def to_s
      return super.to_s + " casilla: " + @valor +  " "
    end
    
  end
end